### ISSUE TITLE
[iOS] The following Swift pods cannot yet be integrated as static libraries

### ISSUE THREAD

**Issue by waterdoo** [RevenueCat/react-native-purchases#183]
**Describe the bug**
After running pod install, I get this error. Reproduced this issue with another on our team.

```
[!] The following Swift pods cannot yet be integrated as static libraries:

The Swift pod `RNPurchases` depends upon `React`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.
```

**Environment**
Platform: iOS
SDK version: 3.4.2
React Native version: 0.59.9
Xcode version: 11.2.1
Cocoapods: tried on 1.6.0.beta.2 and 1.9.3

**Steps to reproduce**

Go through the installation steps.
After running pod install, that error appears.
After attempting to set `:modular_headers => true` per error suggestion, other packages break and Xcode is unable to build.
Also uncommenting use_frameworks in the Podfile but that broke other packages also.

Podfile below:
```
# Uncomment the next line to define a global platform for your project
platform :ios, '9.0'

target 'project' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'Core',
    'CxxBridge',
    'DevSupport',
    'RCTText',
    'RCTNetwork',
    'RCTWebSocket',
    'RCTAnimation',
    'RCTImage',
    # Add any other subspecs you want to use in your project
  ]

  pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

  # Auth0 Lock
  pod 'Lock', '~> 1.26'
  pod 'Lock/TouchID'
  pod 'Lock/SMS'
  pod 'Lock/Email'
  pod 'Lock/Safari'

  pod 'Bolts'
  pod 'FBSDKCoreKit'
  pod 'FBSDKShareKit'
  pod 'FBSDKLoginKit'

  pod "Lock-Facebook", "~> 2.1"


  # react-native-amplitude-analytics
  pod 'Amplitude-iOS', '~> 4.3.1'

  # Below fixes error "Cannot read property 'Constants' of undefined" in any component which imports Video
  # https://github.com/react-native-community/react-native-video/issues/147#issuecomment-422912736
  pod 'react-native-video', :path => File.join('../node_modules/react-native-video', 'react-native-video.podspec')


  # Third party deps podspec link
  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'


  pod 'react-native-amplitude-analytics', :path => '../node_modules/react-native-amplitude-analytics'

  pod 'react-native-image-resizer', :path => '../node_modules/react-native-image-resizer'

  pod 'BVLinearGradient', :path => '../node_modules/react-native-linear-gradient'

  pod 'Picker', :path => '../node_modules/react-native-picker'

  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'

  pod 'react-native-youtube', :path => '../node_modules/react-native-youtube'

  pod 'BugsnagReactNative', :path => '../node_modules/bugsnag-react-native'

  pod 'react-native-branch', :path => '../node_modules/react-native-branch'

  pod 'RNFS', :path => '../node_modules/react-native-fs'

  pod 'react-native-image-picker', :path => '../node_modules/react-native-image-picker'

  pod 'react-native-keep-awake', :path => '../node_modules/react-native-keep-awake'

  pod 'react-native-orientation-locker', :path => '../node_modules/react-native-orientation-locker'

  pod 'CodePush', :path => '../node_modules/react-native-code-push'

  pod 'ReactNativeSearchBar', :path => '../node_modules/react-native-search-bar'

  pod 'RNScreens', :path => '../node_modules/react-native-screens'

  pod 'RNReanimated', :path => '../node_modules/react-native-reanimated'

  pod 'RNGestureHandler', :path => '../node_modules/react-native-gesture-handler'

  pod 'RNPurchases', :path => '../node_modules/react-native-purchases'


end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    # Fix for Xcode Archive Error: 'duplicate symbols for architecture armv7'
    # https://github.com/facebook/react-native/issues/12814
    # Causes warnings "[!] [Xcodeproj] Generated duplicate UUIDs:"
    if target.name == "React"
      target.remove_from_project
    end
    # Fix for Xcode Archive Error - End
  end
end
```

**Comment by aboedo** [comment]
in order to fix the issues in https://github.com/RevenueCat/react-native-purchases/pull/191, we're moving away from depending on the `React` pod and instead depending on `React-Core`. We also added the `DEFINES_MODULE` setting, so those together might help here.
However, I think some of `React-Core`'s dependencies might actually still present issues, from my testing. I've been able to make them work by adding `:modular_headers => false` to the trouble ones, in my case `DoubleConversion`, `glog`, and `Folly` (it looks like they're defined in your `Podfile` as well)
